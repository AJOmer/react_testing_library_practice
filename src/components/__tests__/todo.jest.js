import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Todo from '../Todo';

afterEach(() => {
    cleanup();
})

test('should render non-completed todo component', () => {
    const todo = { id: 1, title: 'Wash butt', completed: false}
    render(<Todo todo = {todo}/>);
    const todoElement = screen.getByTestId( 'todo-1');
    expect(todoElement).toBeInTheDocument();
    expect(todoElement).toHaveTextContent('Wash butt');
    expect(todoElement).not.toContainHTML('<strike>');
})

test('should render completed todo component', () => {
    const todo = {id: 2, title: 'network linkedin', completed: true };
    render(<Todo todo = {todo}/>);
    const todoElement = screen.getByTestId( 'todo-2');
    expect(todoElement).toBeInTheDocument();
    expect(todoElement).toHaveTextContent('network linkedin');
    expect(todoElement).toContainHTML('</strike>');
})

test('matches snapshot', () => {
    const todo = {id: 1, title: 'apply for jobs', completed: false};
    const tree = renderer.create(<Todo todo={todo}/>).toJSON();
    expect(tree).toMatchSnapshot();
})