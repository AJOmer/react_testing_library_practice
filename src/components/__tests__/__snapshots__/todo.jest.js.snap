// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="todo-1"
>
  <h1>
     
    apply for jobs
  </h1>
</div>
`;
